src/anvil.js

```diff
@@ -36,7 +39,17 @@ module.exports = (mcVersion) => {
 
     async loadRaw (x, z) {
       const region = await this.getRegion(x, z)
-      return region.read(x & 0x1F, z & 0x1F)
+      const data = await region.read(x & 0x1F, z & 0x1F)
+      if (mcVersion === 'auto') { // parse version
+        const version = nbt.simplify(data)?.DataVersion ?? -380 // -380 = 1.8
+        mcVersion = unversionedMcData.versions.pc.find(o => o.dataVersion === version).majorVersion
+      }
+      if (nbtChunkToPrismarineChunk === null && prismarineChunkToNbt === null) {
+        const chunk = require('./chunk')(mcVersion)
+        nbtChunkToPrismarineChunk = chunk.nbtChunkToPrismarineChunk
+        prismarineChunkToNbt = chunk.prismarineChunkToNbt
+      }
```

Comment:
This will not work if world was been converted from older versions (because chunks may have different `DataVersion`, since chunk convertion happens lazily).

Reply:
I can suggest set `prismarineChunkToNbt` to latest detected version only, or use `Level.dat` file beside region files.